mathics/doc/common_doc.py

```diff
@@ -596,6 +734,196 @@ def __str__(self) -> str:
+                            doctest.section = self.title
+                            doctest.key = (
+                                doctest.part,
+                                doctest.chapter,
+                                doctest.section,
+                                index,
+                            )
+                            index += 1
+                            new_items.append(doctest)
+
+                            if len(new_items) > 0:
+                                self.tests = Tests(
+                                    doctest.part,
+                                    doctest.chapter,
+                                    doctest.section,
+                                    new_items,
+                                )
+                                yield self.tests
+                    return
+
+                for subsection in section.subsections:
+                    # FIXME we are omitting the section title here...
+                    if not subsection.installed:
+                        continue
+                    for doctests in subsection.items:
+                        yield doctests.get_tests()
+                        pass
+                    pass
+
+            if len(self.items) > 0:
+                self.tests = Tests(
+                    doctest.part, doctest.chapter, doctest.section, self.items
+                )
+                yield self.tests
+            pass
+
+        elif self.tests is None:
+            # Section that is not under a Guide Section.
+            # The Tutorials Part works this way.
+            # DRY with above
+            assert len(self.items) == 0
+            index = 1
+            for doctest in self.xml_doc.get_tests():
+                if isinstance(doctest, DocTest):
+                    doctest.index = index
+                    doctest.chapter = self.chapter.title
+                    doctest.part = self.chapter.part.title
+                    doctest.section = self.title
+                    doctest.key = (
+                        doctest.part,
+                        doctest.chapter,
+                        doctest.section,
+                        index,
+                    )
+                    index += 1
+                    self.items.append(doctest)
+                    pass
+                pass
+
+            if len(self.items) > 0:
+                self.tests = Tests(
+                    doctest.part, doctest.chapter, doctest.section, self.items
+                )
+                yield self.tests
+            pass
+        else:
+            assert False
+            yield self.tests
+
+
+# This stores a pointer to the Mathics3 Module portion of docs In interactive
+# sessions, Mathics3 Modules get loaded on user demand. We need to update that
+# portion of the documenation separately. Storing a pointer here allows us to
+# retrive this portion
+mathics3_module_part: Optional[DocPart] = None
```

Comment:
IMHO, this does not belong here. On the one hand, once you have the ``documentation`` object (of the  suitable subclass) can be accessed through
```documentation.parts_by_slug.get("mathics3-modules", None)```. 
Notice that the part could not exist if in the session no modules were loaded.